subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ProgramSource.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.execution
+
+import org.gradle.internal.hash.Hashing
+
+
+data class ProgramSource(val path: String, val contents: ProgramText) {
+
+    constructor(path: String, contents: String) : this(path, text(contents))
+
+    val text: String
+        get() = contents.text
+
+    fun map(transform: (ProgramText) -> ProgramText) =
+        ProgramSource(path, transform(contents))
+}
+
+
+/**
+ * Normalised program text guaranteed to use `\n` as the only line terminator character.
+ */
+data class ProgramText private constructor(val text: String) {
+
+    companion object {
+
+        fun from(string: String) =
+            ProgramText(string.replace("\r\n", "\n"))
```

Comment:
You only care about Windows an Unix systems here? You don't care about the other possible line terminators correct?
https://en.wikipedia.org/wiki/Newline#Representations_in_different_character_encoding_specifications

Reply:
Correct.